AXX-crl-enhancements.md

```diff
@@ -0,0 +1,220 @@
+AXX: Certificate Revocation List Enhancements
+----
+* Author(s): Gregory Cooke (@gtcooke94, @gregorycooke)
+* Approver: TODO 
+* Status: TODO
+* Implemented in: C, Go
+* Last updated: 2023-07-25
+* Discussion at: TODO
+
+## Abstract
+
+gRPC has an MVP implementation for CRLs (Certificate Revocation Lists). We wish to provide a more generic interface. In addition, we will provide two concrete implementations of this interface similar to Credential Reloading - a static provider and a directory reloader.
+
+
+## Background
+
+[RFC 5280](https://datatracker.ietf.org/doc/html/rfc5280) defines the desired behavior for Certificate Revocation Lists (CRLs). gRPC has an MVP implementation of this RFC for CRL handling in several languages, each with their own flavors - for example, C++ uses OpenSSL, whereas Go uses an implementation written in Go (and the stack in Java is significatly different). There are mismatches and deficiencies between the current Go and C++ implementations. Providing CRL reloading as a feature allows for more internal cohesion with the ideas behind credential reloading.
```

Comment:
I'd drop 'There are mismatches and deficiencies between the current Go and C++ implementations.' statement as the next one looks sufficient for me

Reply:
Done